systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -337,10 +337,11 @@
 ret{1} = zeros(16, nq);
 for i = 2 : nb
   body = bodies(i);
-  T_body_to_parent = T{body.parent} \ T{i};
+  %T_body_to_parent = T{body.parent} \ T{i};
+  T_body_to_parent = T{body.parent}^-1 * T{i};
```

Comment:
True - this is one of the first changes I made, but it also made the test
work.  However, it is possible that it worked because of a special case.  I
am unsure, I would have to investigate more carefully on more plants.

If this isn't general, this is something very core, just the simple
kinematics.  If this can't work, then I agree it would probably be best to
scrap the msspoly and consider something symbolic.

-Andy S.


On Thu, Jul 30, 2015 at 11:34 PM, Twan Koolen <notifications@github.com>
wrote:

> In systems/plants/@RigidBodyManipulator/doKinematics.m
> <https://github.com/RobotLocomotion/drake/pull/1212#discussion_r35944492>:
>
> > @@ -337,10 +337,11 @@
> >  ret{1} = zeros(16, nq);
> >  for i = 2 : nb
> >    body = bodies(i);
> > -  T_body_to_parent = T{body.parent} \ T{i};
> > +  %T_body_to_parent = T{body.parent} \ T{i};
> > +  T_body_to_parent = T{body.parent}^-1 * T{i};
>
> but it doesn't support ^-1 either, does it? In any case, the results can't
> be right in general, as you're outside of the class of polynomials.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1212/files#r35944492>.
>